### PR TITLE
Implement `maxFailures` and warning messages

### LINKED ISSUE #34044: Implement `maxFailures` in E2E tests to limit failures and size of Allure reports
### Describe the solution you'd like

Currently, E2E test runs that have too many failures generate huge Allure test reports, and take a very long time to complete like so:

Screen Shot 2022-07-19 at 8 23 47 PM

To address this, we can use Playwright's [`maxFailures`](https://playwright.dev/docs/api/class-testconfig#test-config-max-failures) configuration to implement a fail-fast mechanism to save time on testing a faulty build, and to prevent the Allure reports from getting huge.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rodelgc**
### Changes proposed in this Pull Request:

Currently, E2E test runs that have too many failures generate huge Allure test reports, and take a very long time to complete like so:

Screen Shot 2022-07-19 at 8 23 47 PM

To address this, we can use Playwright's [`maxFailures`](https://playwright.dev/docs/api/class-testconfig#test-config-max-failures) configuration to implement a fail-fast mechanism to save time on testing a faulty build, and to prevent the Allure reports from getting huge.

Closes https://github.com/woocommerce/woocommerce/issues/34044.

### How to test the changes in this Pull Request:

1. In this pull request or in a duplicate one, edit a few Playwright E2E tests to produce more than 3 failures. For example, you may edit the locator in this test to a wrong one (say, from `#woocommerce-layout__primary` to `#wrong_locator`) so that it would produce more than 3 failures:
    https://github.com/woocommerce/woocommerce/blob/049c0d17cc25179cc0581311aec827897940f429/plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics.spec.js#L29
2. Push this change and wait for the "Run tests against PR" workflow to finish.
3. Verify that the Test Results Summary shows:
    - A "Warning" section that says there were:
        - failed and broken tests
        - incomplete E2E test run
    - The table shows the "Total" for E2E tests is less than 186 (the total number of E2E tests we have currently)
    - The table shows the duration of E2E tests to be just several minutes, not reaching almost an hour.
    - The `e2e-test-report---pr-33977` artifact is only around 50 MB, no longer reaching hundreds of MB's like before.
    Screen Shot 2022-07-22 at 8 29 28 PM
5. Revert the last change to remove the intentional failures we added.
6. This time, edit a few tests so that only 1 or 2 tests would fail. For example, you may edit the expected title in this line so that the E2E test run will produce only 1 failure.
    https://github.com/woocommerce/woocommerce/blob/e2e/pr-fail-fast/plugins/woocommerce/tests/e2e-pw/tests/basic.spec.js#L32
7. Push this change and wait again for the " Run tests against PR" workflow to complete.
8. Verify that the Test Results Summary shows only 1 warning: the one about failures and broken tests. It no longer shows the warning about incomplete E2E test run.
9. Verify that the table shows the "Total" for E2E tests to be 186, which means that all E2E tests were executed.
    Screen Shot 2022-07-22 at 7 51 13 PM

11. Revert the previous change so that no more failures will occur.
12. Push this change and wait again for the " Run tests against PR" workflow to complete.
13. Verify that all E2E tests ran (186 currently) based on the table in the Test Results Summary.
14. Verify that no more warnings were displayed in the Test Results Summary.
     Screen Shot 2022-07-22 at 8 46 50 PM

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .github/workflows/scripts/prepare-test-summary.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const {
 	E2E_PPTR_SUMMARY_PATH,
 	SHA,
 	PR_NUMBER,
+	E2E_GRAND_TOTAL,
 } = process.env;
```
Comment (alopezari) [inline .github/workflows/scripts/prepare-test-summary.js:11]: This file has become a big one (+250 code lines), do you think we can split it somehow @rodelgc? 

Perhaps we could have:
- One file with the functions to retrieve stats (`getAllureSummaryStats`, `getPuppeteerStats`...)
- Another file to prepare the static attributes of the report (for example `createAPITableRow`, `createE2ETableRow`, `addSummaryFooter`...)
- Another file to handle warnings/errors

I don't have enough knowledge of this code so I'm just thinking out loud, but would something like this make sense? Of course, we could follow another criteria to split the code if it's more appropriate.

**Comment by alopezari** [comment]
By the way, great work on this @rodelgc! This is a must-have feature in every test framework, so it's amazing that we have it now in ours. Thank you very much! 🙌

**Review by alopezari — DISMISSED** [review]

**Comment by lanej0** [comment]
Just pinging @woocommerce/masamune for awareness.  I think @Luc45 was mentioning something similar to this at one point (but I might be mis-remembering)

**Comment by tammullen** [comment]
@rodelgc looks like this just needs the conflicts resolved

**Comment by rodelgc** [comment]
Seems like some of the conflicts were due to some files being relocated in earlier PR's. I'll resolve the merge conflicts within this week. Converting this back to draft for the meantime.

**Comment by rodelgc** [comment]
Done resolving the merge conflicts! And ready for another round of review.

**Review by lanej0 — APPROVED** [review]

**Pull request merged**